src/Nethermind/Nethermind.Trie/BatchedTrieVisitor.cs

```diff
@@ -293,10 +300,12 @@ private void BatchedThread()
 
                 cur.ResolveKey(_resolver, false);
 
-                SmallTrieVisitContext ctx = currentBatch[i].Item2;
+                SmallTrieVisitContext ctx = currentBatch[i].Item3;
 
                 if (cur.FullRlp.IsNotNull) continue;
-                if (cur.Keccak is null) throw new TrieException($"Unable to resolve node without Keccak. ctx: {ctx.Level}, {ctx.ExpectAccounts}, {ctx.IsStorage}, {ctx.BranchChildIndex}");
+                if (cur.Keccak is null)
+                    throw new TrieException(
```

Comment:
Move this Throw out to a local function so it the method doesn't have all the extra junk in it for newing up the exception, interplolating the string and creating the string. Increases the amount of cold code that the method cpu has to fill its instruction cache with

Reply:
Fixed.